metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -2505,22 +2529,24 @@ class MetalsLspService(
 
   private def onBuildChangedUnbatched(
       paths: Seq[AbsolutePath]
-  ): Future[Unit] = {
-    val changedBuilds = paths.flatMap(buildTools.isBuildRelated)
-    tables.buildTool.selectedBuildTool() match {
-      // no build tool and new added
-      case None if changedBuilds.nonEmpty =>
-        scribe.info(s"Detected new build tool in $path")
-        fullConnect()
-      // used build tool changed
-      case Some(chosenBuildTool) if changedBuilds.contains(chosenBuildTool) =>
-        slowConnectToBuildServer(forceImport = false).ignoreValue
-      // maybe new build tool added
-      case Some(chosenBuildTool) if changedBuilds.nonEmpty =>
-        onBuildToolsAdded(chosenBuildTool, changedBuilds)
-      case _ => Future.unit
+  ): Future[Unit] =
+    if (willGenerateBspConfig.get().nonEmpty) Future.unit
```

Comment:
I can't use `isConnecting` flag since it's raised after `.bsp/<name>.json` is created. I could raise it earlier but then I'd have to do it in many places to have it pair its name.